tests/importer/importer.cpp

```diff
@@ -0,0 +1,762 @@
+    importer->resolveImports(modelB, resourcePath("importer/"));
+    importer->resolveImports(modelC, resourcePath("importer/"));
+
+    EXPECT_EQ(size_t(1), importer->libraryCount());
+}
+
+TEST(Importer, addModelToLibrary)
+{
+    // This test shows how a model instance can be manually added to the import library by the user,
+    // and will be used to resolve imports, rather than from an external file/URL.
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("importer/generic_no_source.cellml"));
+    auto importer = libcellml::Importer::create();
+
+    auto sourceModel = libcellml::Model::create("source");
+    sourceModel->addComponent(libcellml::Component::create("a"));
+    sourceModel->addComponent(libcellml::Component::create("b"));
+    sourceModel->addComponent(libcellml::Component::create("c"));
+    sourceModel->addComponent(libcellml::Component::create("d"));
+
+    // Add a model manually to the library, including the URL that it will replace in future imports.
+    EXPECT_TRUE(importer->addModel(sourceModel, "i_dont_exist.cellml"));
+
+    // Can't add to the same URL key more than once.
+    EXPECT_FALSE(importer->addModel(sourceModel, "i_dont_exist.cellml"));
+
+    importer->resolveImports(model, "");
+
+    EXPECT_EQ(size_t(0), importer->issueCount());
+    EXPECT_FALSE(model->hasUnresolvedImports());
+}
+
+TEST(Importer, replaceModel)
+{
+    // This test shows how a model instance can be manually added to the import library by the user,
+    // and will be used to resolve imports, rather than from an external file/URL.
+    auto parser = libcellml::Parser::create();
+    auto model = parser->parseModel(fileContents("importer/generic_no_source.cellml"));
+    auto importer = libcellml::Importer::create();
+
+    auto wrongSourceModel = libcellml::Model::create("wrong");
+
+    auto rightSourceModel = libcellml::Model::create("source");
+    rightSourceModel->addComponent(libcellml::Component::create("a"));
+    rightSourceModel->addComponent(libcellml::Component::create("b"));
+    rightSourceModel->addComponent(libcellml::Component::create("c"));
+    rightSourceModel->addComponent(libcellml::Component::create("d"));
+
+    // Add a model manually to the library, including the URL that it will replace in future imports.
+    EXPECT_TRUE(importer->addModel(wrongSourceModel, "i_dont_exist.cellml"));
+    // Can't add to the same URL key more than once.
+    EXPECT_FALSE(importer->addModel(rightSourceModel, "i_dont_exist.cellml"));
+
+    // Use replaceModel instead.
+    EXPECT_FALSE(importer->replaceModel(rightSourceModel, "not_in_library"));
+    EXPECT_TRUE(importer->replaceModel(rightSourceModel, "i_dont_exist.cellml"));
+
+    importer->resolveImports(model, "");
+
+    EXPECT_EQ(size_t(0), importer->issueCount());
+    EXPECT_FALSE(model->hasUnresolvedImports());
+}
+
+TEST(Importer, getNonexistentModel)
+{
+    // For coverage.
```

Comment:
This test actually serves a purpose, so would remove the comment. If not, then the name of the test should be changed to `coverage`.

Reply:
65c5656e